sparkmagic/sparkmagic/tests/test_sparkstorecommand.py

```diff
@@ -0,0 +1,241 @@
+import pandas as pd
+from pandas.util.testing import assert_frame_equal
+
+import sparkmagic.utils.configuration as conf
+from sparkmagic.utils.constants import LONG_RANDOM_VARIABLE_NAME
+from sparkmagic.livyclientlib.sparkcommand import SparkStoreCommand
+from sparkmagic.livyclientlib.command import Command
+from sparkmagic.livyclientlib.exceptions import BadUserDataException
+
+
+def _setup():
+    pass
+    
+
+def _teardown():
+    pass
+
+
+@with_setup(_setup, _teardown)
+def test_to_command_pyspark():
+    variable_name = "var_name"
+    sparkcommand = SparkStoreCommand("Code", variable_name)
+    sparkcommand._pyspark_command = MagicMock(return_value=MagicMock())
+    sparkcommand.to_command("pyspark", variable_name)
+    sparkcommand._pyspark_command.assert_called_with(variable_name)
+
+
+@with_setup(_setup, _teardown)
+def test_to_command_pyspark3():
+    variable_name = "var_name"
+    sparkcommand = SparkStoreCommand("Code", variable_name)
+    sparkcommand._pyspark_command = MagicMock(return_value=MagicMock())
+    sparkcommand.to_command("pyspark3", variable_name)
+    sparkcommand._pyspark_command.assert_called_with(variable_name, False)
+
+@with_setup(_setup, _teardown)
+def test_sparkstorecommand_initializes():
+    variable_name = "var_name"
+    code = "HERE IS MY SPARK CODE ON RDDs"
+    samplemethod = "take"
+    maxrows = 120
+    samplefraction = 0.6
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod, maxrows, samplefraction)
+    assert_equals(sparkcommand.code, code)
+    assert_equals(sparkcommand.samplemethod, samplemethod)
+    assert_equals(sparkcommand.maxrows, maxrows)
+    assert_equals(sparkcommand.samplefraction, samplefraction)
+
+
+@with_setup(_setup, _teardown)
+def test_sparkstorecommand_loads_defaults():
+    defaults = {
+        conf.default_samplemethod.__name__: "sample",
+        conf.default_maxrows.__name__: 419,
+        conf.default_samplefraction.__name__: 0.99,
+    }
+    conf.override_all(defaults)
+    variable_name = "var_name"
+    code = "hugedataset.collect()"
+    sparkcommand = SparkStoreCommand(code, variable_name)
+    assert_equals(sparkcommand.code, code)
+    assert_equals(sparkcommand.samplemethod, defaults[conf.default_samplemethod.__name__])
+    assert_equals(sparkcommand.maxrows, defaults[conf.default_maxrows.__name__])
+    assert_equals(sparkcommand.samplefraction, defaults[conf.default_samplefraction.__name__])
+
+@with_setup(_setup, _teardown)
+@raises(BadUserDataException)
+def test_sparkstorecommand_rejects_bad_data():
+    variable_name = "var_name"
+    code = "hugedataset.collect()"
+    samplemethod = "foo"
+    _ = SparkStoreCommand(code, variable_name, samplemethod)
+
+
+@with_setup(_setup, _teardown)
+def test_pyspark_livy_sampling_options():
+    variable_name = "abc"
+    code = "hugedataset.collect()"
+    sparkcommand = SparkStoreCommand(code, variable_name, samplemethod='take', maxrows=120)
```

Comment:
We are missing a test for when sample method is `None` and it should do a `collect` for all 4 languages